gtk4-macros/src/lib.rs

```diff
@@ -261,3 +261,69 @@ pub fn template_callbacks(attr: TokenStream, item: TokenStream) -> TokenStream {
         Err(_) => abort_call_site!(template_callbacks_attribute::WRONG_PLACE_MSG),
     }
 }
+
+/// Attribute macro for declaring GTK tests.
+///
+/// Wraps the standard Rust [`#[test]`] attribute with setup logic for GTK. All tests that call
+/// into GTK must use this attribute. This attribute can also be used on asynchronous functions;
+/// the asynchronous test will be run on the main thread context.
+///
+/// # Technical Details
+///
+/// GTK is a single-threaded library, so Rust's normal multi-threaded test behavior cannot be used.
+/// The `#[gtk::test]` attribute creates a main thread for GTK and runs all tests on that thread.
+/// This has the side effect of making all tests run serially, not in parallel.
+///
+/// [`#[test]`]: (https://doc.rust-lang.org/std/prelude/v1/macro.test.html)
+///
+/// # Example
+///
+/// ```no_run
+/// use gtk::prelude::*;
+///
+/// #[gtk::test]
+/// fn test_button() {
+///     let button = gtk::Button::new();
+///     button.activate();
+/// }
+/// ```
+#[proc_macro_attribute]
+#[proc_macro_error]
+pub fn test(_attr: TokenStream, item: TokenStream) -> TokenStream {
+    use proc_macro_error::abort_call_site;
+    use quote::quote;
+
+    match syn::parse::<syn::ItemFn>(item) {
```

Comment:
Might want to add some custom error if the function takes parameters or returns something

Reply:
That is all handled already by `#[test]`, we can just fall back on that logic i.e. the handling of return values that implement `std::process::Termination`.